rotkehlchen/chain/evm/decoding/stakedao/utils.py

```diff
@@ -20,24 +26,35 @@
 SUPPORTED_STAKEDAO_STRATEGIES: Final = {'curve', 'balancer', 'pendle', 'pancakeswap', 'angle'}
 
 
-def query_stakedao_gauges() -> None:
+def query_stakedao_gauges(evm_inquirer: 'EvmNodeInquirer') -> None:
     """Query StakeDAO gauges for lockers and strategies from the API and cache them."""
-    all_gauges: dict[ChainID, set] = defaultdict(set)
-
+    all_gauges: set[tuple[ChecksumEvmAddress, ChecksumEvmAddress]] = set()
     try:  # Query lockers
         response_data = request_get_dict(url=f'{STAKEDAO_API}/lockers')
         for item in response_data['parsed']:
+            if item['chainId'] != evm_inquirer.chain_id.serialize():
+                continue
+
             try:
-                all_gauges[ChainID.deserialize(item['chainId'])].add(deserialize_evm_address(item['modules']['gauge']))
+                all_gauges.add((
+                    deserialize_evm_address(item['modules']['gauge']),
+                    deserialize_evm_address(item['sdToken']['address']),  # underlying token
+                ))
             except (KeyError, DeserializationError) as e:
                 log.warning(f'Skipping stakedao locker entry {item} due to {e!s}')
                 continue
 
         for protocol in SUPPORTED_STAKEDAO_STRATEGIES:  # Query strategies
             response_data = request_get_dict(url=f'{STAKEDAO_API}/strategies/{protocol}')
             for item in response_data['deployed']:
+                if item['chainId'] != evm_inquirer.chain_id.serialize():
+                    continue
```

Comment:
you are serializing in the nested for loop. And you use it also before in line 35. Serialize one and store in a variable